src/main/java/io/vertx/ext/mail/impl/SMTPSendMail.java

```diff
@@ -187,57 +202,177 @@ private void dataCmd() {
   }
 
   private void sendMaildata() {
```

Comment:
Shouldn't this method become async? it calls an async method so if you're not on the right timing you might get the state wrong or be unable to catch any eventual error.

Reply:
What I understand on this method is that it will be called only once after getting a succeed result of writing `DATA`  in method: `dataCmd()`.

I need to make some modification to make this async.